### PR TITLE
Add more doc to MediaQuery padding

### PR DESCRIPTION
**Pull request by xster**

### CODE REVIEW

**Review by tvolkert — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/widgets/media_query.dart [hunk 1]
```diff
@@ -89,6 +89,11 @@ class MediaQueryData {
   /// system UI (such as the system notification area), or or physical
   /// intrusions in the display (e.g. overscan regions on television screens or
   /// phone sensor housings).
+  ///
+  /// If you consumed this padding (e.g. by building a widget that envelops or
+  /// accounts for this padding in its layout), you should remove this padding
```
Comment (cbracken) [inline packages/flutter/lib/src/widgets/media_query.dart]: This is *mostly* true, but there are cases, such as translucent widgets, where you both want to account for the padding and allow children to account for it. Maybe make the language less absolute or add something like 'if your widget accounts for this padding such that child widgets do not need to' (though ideally with better words than those...)

File: packages/flutter/lib/src/widgets/media_query.dart [hunk 2]
```diff
@@ -89,6 +89,11 @@ class MediaQueryData {
...
+  /// for subsequent descendents in the widget tree by inserting a new
+  /// [MediaQuery] widget using the [MediaQuery.removePadding] factory.
```
Comment (cbracken) [inline packages/flutter/lib/src/widgets/media_query.dart]: We should suggest `SafeArea` here for the straightforward case where you just need to apply padding and remove it.

File: packages/flutter/lib/src/widgets/media_query.dart (hunk 1 above)
Comment (xster) [reply packages/flutter/lib/src/widgets/media_query.dart]: Added more comments

File: packages/flutter/lib/src/widgets/media_query.dart (hunk 2 above)
Comment (xster) [reply packages/flutter/lib/src/widgets/media_query.dart]: Ah man, TIL. I totally replicated that work in some places.

**Comment by cbracken** [comment]
This looks great! Thanks!

**Pull request merged**